portfolio/src/main/java/com/google/sps/servlets/UserServlet.java

```
@@ -0,0 +1,89 @@
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+
+import com.google.gson.Gson;
+import com.google.gson.annotations.SerializedName;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.users.User;
+
+/** Servlet that handles user authentication. */
+@WebServlet("/user")
+public class UserServlet extends HttpServlet {
+
+  private class LoginStatus {
+    private boolean logged_in;
+    private String url;
+    private String nickname;
+
+    public LoginStatus(boolean logged_in, String url, String nickname) {
+      this.logged_in = logged_in;
+      this.url = url;
+      this.nickname = nickname;
+    }
+  }
+
+  String makeJSONStatus(boolean logged_in, String url, String nickname) {
+    Gson gson = new Gson();
+    String json = gson.toJson(new LoginStatus(logged_in, url, nickname));
+    return json;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+
+    boolean logged_in = userService.isUserLoggedIn();
+    String url = (logged_in)
+        ? userService.createLogoutURL("/")
+        : userService.createLoginURL("/");
+
+    User user = userService.getCurrentUser();
+    String id = (user != null) ? user.getUserId() : "";
```

Will this work if id gets set to ""? Should we return an error message if user.getUserId() is empty?